### ISSUE TITLE
TypeError: React.createContext is not a function

### ISSUE THREAD

**Issue by brianyang** [eldh/statext#1]
I used this in an apollo application and get this error. 

the code looks like this
```
withSharedState(compose(graphql)(ComponentClass))
```
is this the right syntax? have any idea what this means? 
```
/Users/?/node_modules/statext/dist/index.js:12
var _createContext = React.createContext('foo'),
                           ^
TypeError: React.createContext is not a function
    at Object.<anonymous> (/Users/?/node_modules/statext/dist/index.js:12:28)
    at Module._compile (module.js:573:30)
    at Object.Module._extensions..js (module.js:584:10)
    at Module.load (module.js:507:32)
    at tryModuleLoad (module.js:470:12)
    at Function.Module._load (module.js:462:3)
    at Module.require (module.js:517:17)
    at require (internal/module.js:11:18)
```

**Comment by brianyang** [comment]
duh won't work because the wrapped provider is wrong

**Comment by eldh** [comment]
@matteom4rongiu Could you provide some more context on where you are getting this error?  Make sure you are using React 16.3 or higher, since Statext requires the [new context api](https://reactjs.org/blog/2018/03/29/react-v-16-3.html) to work.

**Comment by matteom4rongiu** [comment]
Ok thanks, I solved updating React to 16.4.1:
I deleted modules, than i run "npm upgrade react@^16.4.1" and finally npm update.
